library/core/src/main/java/com/google/android/exoplayer2/text/Cue.java

```diff
@@ -98,11 +101,33 @@
   public static final int TEXT_SIZE_TYPE_ABSOLUTE = 2;
 
   /**
-   * The cue text, or null if this is an image cue. Note the {@link CharSequence} may be decorated
+   *   {@link #maxWindowColumnCount} or {@link #maxWindowRowCount} unset
+   */
+  public static final int WINDOW_ROW_COLUMN_COUNT_UNSET = 0;
+
+  /**
+   * Value for {@link #windowBorderType}
+   */
+  public static final int BORDER_AND_EDGE_TYPE_NONE = 0;
+  public static final int BORDER_AND_EDGE_TYPE_RAISED = 1;
+  public static final int BORDER_AND_EDGE_TYPE_DEPRESSED = 2;
+  public static final int BORDER_AND_EDGE_TYPE_UNIFORM = 3;
+  public static final int BORDER_AND_EDGE_TYPE_SHADOW_LEFT = 4;
+  public static final int BORDER_AND_EDGE_TYPE_SHADOW_RIGHT = 5;
+
+
+  /**
+   * The list of pair of cue text and column offsets. Note the {@link CharSequence} may be decorated
+   * with styling spans.
+   */
+  public final List<Pair<Integer, CharSequence>> textRows;
```

Comment:
Does this effectively duplicate the text in every `Cue`? That doesn't seem ideal from a modelling point of view. Given CEA uses monospace fonts, why can't columns be handled using spaces at the start of each line in `text`?

Reply:
It does not duplicate the text.  Either text field is set or textRows is set, not both. 
Earlier text field used to contain a 4 line text. But now textRows will have 4 elements, each having one line of text and associated column offset. 

In fact, initially we did implement column offset by inserting a simple space. The problem with this approach is that space draws the text background, which does not match the expected reference output when the window color is transparent.